### ISSUE TITLE
Trainer.add_argparse_args bool type 

### ISSUE THREAD

**Issue by fschlatt** [PyTorchLightning/pytorch-lightning#1570]
## 🐛 Bug

The boolean arguments added using `Trainer.add_argparse_args` always evaluate to True. This is caused by the following lines of the add_argparse_args fucntion:
```python
if isinstance(allowed_type, bool):
    def allowed_type(x):
         return bool(distutils.util.strtobool(x))
```
Because `allowed_type` is the actual data type and not an instance of bool. `isinstance(bool, bool)` is equal to `False`. `bool is bool` is equal True

**Comment by fschlatt** [comment]
I just found the `Trainer.add_argparse_args` clashes with the auto_lr_find argument as auto_lr_find can be either bool or str. I would propose removing the option to provide a string to auto_lr_find. If I understand correctly then the str option is only there to specify a separate learning rate using argparse anyway. On top of being confusing you could just use hparams.lr and set auto_lr_find to False which is more straightforward
